src/api.js

```diff
@@ -464,56 +511,104 @@ var remoteAuthentication = function(options, entry) {
           return false;
         }
 
-        // If we're not given an array, we assume it's a set of parameters that
-        // must be used to parameterize the original scopesets
-        if (!(scopesets instanceof Array)) {
-          var params = scopesets;
-          scopesets = _.cloneDeepWith(entry.scopes, function(scope) {
+        req.hasAuthed = false;
```

Comment:
I feel like this should be set earlier -- although I guess undefined is falsey too so it's not too important.